### PR TITLE
feat(fields): Created directive for controlling required fields UI [SDNTB-21]

### PR DESCRIPTION
**Pull request by pavlovicnemanja**
SDNTB-21

### CODE REVIEW

**Comment by ioanpocol** [comment]
:+1:

File: scripts/superdesk/ui/ui.js [hunk 1]
```diff
@@ -1425,6 +1425,52 @@
...
+                if (_.has(elem[0].attributes, 'sd-meta-terms')) {
+                    $timeout(function () {
+                        if (elem.find('.terms-list').html()) {
+                            validate(elem.find('.terms-list').html());
+                        }
+
+                        elem.find('.dropdown-toggle').on('blur', function () {
+                            validate(elem.find('.terms-list').html());
+                        });
+                    }, 5000);
```
Comment (petrjasek) [inline scripts/superdesk/ui/ui.js]: you can add false here..

File: scripts/superdesk/ui/ui.js [hunk 2]
```diff
@@ -1425,6 +1425,52 @@
...
+                } else if (_.has(elem[0].attributes, 'sd-text-editor')) {
+                    elem.find('.text-editor').on('blur', function () {
+                        validate($(this).html());
+                    });
+                } else if (elem[0].nodeName === 'INPUT' || elem[0].nodeName === 'TEXTAREA') {
```
Comment (petrjasek) [inline scripts/superdesk/ui/ui.js]: imo would be better if it uses ngModel when possible

**Comment by ioanpocol** [comment]
please rebase

File: scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -1176,6 +1177,21 @@
                     });
                 }
 
+                function validate(orig, item) {
+                    $scope.error = {};
+                    angular.forEach(_.extend(orig, item), function (value, key) {
+                        if (value && value !== '' && value !== null) {
```
Comment (petrjasek) [inline scripts/superdesk-authoring/authoring.js]: fyi if value is `true` it can't be `''` or `null` so those checks are extra

**Comment by petrjasek** [comment]
not sure why are you using `ngRequired` which works with `ngModel`, it should probably have different name

**Comment by ioanpocol** [comment]
:+1:

**Comment by pavlovicnemanja** [comment]
Hey @petrjasek, ng-required read's from content_types and define witch fields are required and witch not.

File: scripts/superdesk-authoring/authoring.js (hunk 3 above)
Comment (pavlovicnemanja) [inline scripts/superdesk-authoring/authoring.js]: Never be 100% sure! :smile:

File: scripts/superdesk-authoring/authoring.js [hunk 4]
```diff
@@ -1176,6 +1177,21 @@
                     });
                 }
 
+                function validate(orig, item) {
+                    $scope.error = {};
+                    angular.forEach(_.extend(orig, item), function (value, key) {
+                        if (value) {
```
Comment (petrjasek) [inline scripts/superdesk-authoring/authoring.js]: btw this function could be ```$scope.error[key] = _.isEmpty(value);```

**Comment by petrjasek** [comment]
:+1:

File: scripts/superdesk-authoring/authoring.js (hunk 4 above)
Comment (pavlovicnemanja) [inline scripts/superdesk-authoring/authoring.js]: :+1: I really like underscore! :smile: tnx

**Pull request merged**